crates/ty_server/src/session.rs

```diff
@@ -488,6 +532,20 @@ impl Session {
             publish_settings_diagnostics(self, client, root);
         }
 
+        if let Some(global_options) = combined_global_options.take() {
+            let global_settings = global_options.into_settings();
+            if global_settings.diagnostic_mode().is_workspace() {
+                for project in self.projects.values_mut() {
+                    if project.db.check_mode() != CheckMode::AllFiles {
+                        project.db.set_check_mode(CheckMode::AllFiles);
+                    }
+                }
```

Comment:
Nit: We could move the `!= CheckMode::AllFiles` into `set_check_mode` method (so that it only sets the new mode if it is different to avoid triggering a new salsa revision)

Reply:
Yeah, I think that's better and now that I realize it it's more inline with some of the other values that are updated (e.g., on `File`).